s3stream/src/main/java/com/automq/stream/s3/S3Storage.java

```diff
@@ -446,48 +446,56 @@ private CompletableFuture<ReadDataBlock> read0(FetchContext context,
         @SpanAttribute long startOffset,
         @SpanAttribute long endOffset,
         @SpanAttribute int maxBytes) {
-        List<StreamRecordBatch> logCacheRecords = deltaWALCache.get(context, streamId, startOffset, endOffset, maxBytes);
-        if (!logCacheRecords.isEmpty() && logCacheRecords.get(0).getBaseOffset() <= startOffset) {
-            return CompletableFuture.completedFuture(new ReadDataBlock(logCacheRecords, CacheAccessType.DELTA_WAL_CACHE_HIT));
+        if (maxBytes <= 0 || startOffset >= endOffset) {
+            return CompletableFuture.completedFuture(new ReadDataBlock(Collections.emptyList(), CacheAccessType.DELTA_WAL_CACHE_HIT));
         }
+
+        StreamRecordBatch first = deltaWALCache.getFirst(streamId, startOffset, endOffset);
```

Comment:
first record release

Reply:
Refactored. Now this method only returns baseOffset of the first offset.